test/data/collection.cpp

```diff
@@ -326,31 +326,35 @@ BOOST_AUTO_TEST_CASE(collection__is_sorted__unsorted__false)
 
 BOOST_AUTO_TEST_CASE(collection__distinct__empty__empty)
 {
-    BOOST_REQUIRE(distinct(data_chunk{}).empty());
+    data_chunk data{};
+    BOOST_REQUIRE(distinct(data).empty());
 }
 
 BOOST_AUTO_TEST_CASE(collection__distinct__single__unchanged)
 {
     const uint8_t expected = 42;
-    BOOST_REQUIRE_EQUAL(distinct(data_chunk{ expected }).front(), expected);
+    data_chunk data{ expected };
+    BOOST_REQUIRE_EQUAL(distinct(data).front(), expected);
```

Comment:
Same with these, they are move overload tests.

Reply:
Sometimes boost macros just don’t like complex expressions.

Reply:
It's not a boost issue.  GCC doesn't support non-const lvalues while Visual Studio accepts them with a default extension.  Maybe I'm missing `std::move` and that will resolve it.  Will investigate.

Reply:
If it’s a move arg it’s an r-value.

Reply:
Sounds like it should be `Collection& distinct(Collection&& list) noexcept` an r-value reference which would avoid the conflict and rationalize the return of the reference being provided.

Reply:
The return would just be 'Collection'